future-apps/payroll/app/src/components/Table/Table.js

```diff
@@ -39,8 +58,92 @@ class Table extends React.Component {
     })
   }
 
+  paginateData = (data) => {
+    const { rowsPerPage } = this.props
+    const { currentPage } = this.state
+    const lastDataIndex = currentPage * rowsPerPage
+    const firstDataIndex = lastDataIndex - rowsPerPage
+    const paginatedData = data.slice(firstDataIndex, lastDataIndex)
+    const emptyRows = rowsPerPage - paginatedData.length
+    const totalPages = Math.ceil(data.length / rowsPerPage)
+
+    if (paginatedData.length === 0) this.setState({ currentPage: 1 })
+
+    return {
+      emptyRows,
+      paginatedData,
+      totalPages
+    }
+  }
+
+  handlePageChange = (pageNumber) => () => {
+    this.setState({
+      currentPage: pageNumber
+    })
+  }
+
+  renderEmptyRows (emptyRows) {
+    const { tableRowHeight } = this.props
+    return emptyRows > 0 && (
+      <TableRow style={{ height: `${tableRowHeight * emptyRows}px` }} />
+    )
+  }
+
+  renderPageButtons (data) {
+    const { currentPage } = this.state
+    const { navigationNextText, navigationPreviousText, rowsPerPage } = this.props
+    const {
+      activeNavigationButton, inactiveNavigationButton,
+      activePageButton, inactivePageButton
+    } = footerStyles
+
+    const pageNumbers = Array(Math.ceil(data.length / rowsPerPage))
+      .fill()
+      .map((_, index) => index + 1)
+
+    const isFirstPage = currentPage === 1
+    const isLastPage = currentPage === pageNumbers.length
+    return (
+      <React.Fragment>
+        <Button
+          name={`pagination-button-previous`}
+          disabled={isFirstPage}
+          mode={isFirstPage ? '' : 'strong'}
+          style={isFirstPage ? inactiveNavigationButton : activeNavigationButton }
+          onClick={this.handlePageChange(currentPage - 1)}
+        >
+          { navigationPreviousText }
+        </Button>
+        {pageNumbers.map((pageNumber) => {
+          const isCurrentPageButton = pageNumber === currentPage
+          const name = `pagination-button-${pageNumber}`
+          return (
+            <Button
```

Comment:
We could have a styled-component for this

Reply:
Page buttons were refactored